sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveTableScanSuite.scala

```diff
@@ -192,4 +192,44 @@ class HiveTableScanSuite extends HiveComparisonTest with SQLTestUtils with TestH
       case p: HiveTableScanExec => p
     }.get
   }
+
+  test("[SPARK-26630] Fix ClassCastException in TableReader while creating HadoopRDD") {
```

Comment:
Added more test cases? like CTAS and then SELECt? INSERT and then SELECT?

Move it to `HiveCatalogedDDLSuite`?